render/src/config.rs

```diff
@@ -1,20 +1,42 @@
 use xsynth_core::{channel::ChannelInitOptions, soundfont::SoundfontInitOptions};
 
+/// Supported audio formats of XSynthRender
 #[derive(PartialEq, Clone, Copy)]
 pub enum XSynthRenderAudioFormat {
     Wav,
 }
 
+/// Options for initializing a new XSynthRender object.
 #[derive(Clone)]
 pub struct XSynthRenderConfig {
+    /// Channel initialization options (same for all channels).
     pub channel_init_options: ChannelInitOptions,
+
+    /// Soundfont initialization options (same for all soundfonts).
     pub sf_init_options: SoundfontInitOptions,
+
+    /// Amount of VoiceChannel objects to be created.
+    /// (Number of MIDI channels)
     pub channel_count: u32,
+
+    /// A vector which specifies which of the created channels (indexes) will be used for drums.
+    /// For example in a conventional 16 MIDI channel setup where channel 10 is used for
+    /// drums, the vector would be set as \[9\] (counting from 0).
     pub drums_channels: Vec<u32>,
+
+    /// Whether or not to use a threadpool to render voices.
     pub use_threadpool: bool,
+
+    /// Whether or not to limit the output audio.
```

Comment:
These should probably be more descriptive in general, as they're very public-facing configuration settings

Reply:
I'll try my best, although for some there is not much more to say as they are pretty self-explanatory like `sample_rate`. Will post all the current changes in a commit now and I'll do the rest after you review everything else.

Reply:
Yeah looks good